### PR TITLE
Gcmon 102

### PR DESCRIPTION
**Pull request by mamcderm-usgs**
@davidmsibley - are you ok with me merging this?

### CODE REVIEW

**Comment by mamcderm-usgs** [comment]
@davidmsibley - as I expected, this was straight-forward once you got me on the right path.  seems to be working but would you take a look when you get a chance to make sure I've got the right idea.  also appreciate any suggestions on how to make the java code better.  and, I still need to add one ancillary column.  thanks.

File: gcmrc-services/src/main/java/gov/usgs/cida/gcmrcservices/nude/AggregatingEndpoint.java [hunk 1]
```diff
@@ -38,6 +41,7 @@
 import java.util.Map.Entry;
 import java.util.Set;
 import java.util.UUID;
+import org.apache.commons.codec.binary.Hex;
```
Comment (davidmsibley) [inline gcmrc-services/src/main/java/gov/usgs/cida/gcmrcservices/nude/AggregatingEndpoint.java]: probably want to clean up these unused imports

File: gcmrc-services/src/main/java/gov/usgs/cida/gcmrcservices/column/ColumnResolver.java [hunk 2]
```diff
@@ -302,8 +302,22 @@ public static String getCustomName(String colName) {
...
+		result.put(parameterCode, new ColumnMetadata(parameterCode, "discharge ice affected", 
+				new ColumnMetadata.SpecEntry(ParameterCode.parseParameterCode(parameterCode), ColumnMetadata.SpecEntry.SpecType.PARAM)));
 		
+		parameterCode = "notes!Stage";
+		result.put(parameterCode, new ColumnMetadata(parameterCode, "stage notes", 
+				new ColumnMetadata.SpecEntry(ParameterCode.parseParameterCode(parameterCode), ColumnMetadata.SpecEntry.SpecType.PARAM)));
+		parameterCode = "iceAffected!Stage";
+		result.put(parameterCode, new ColumnMetadata(parameterCode, "stage ice affected", 
+				new ColumnMetadata.SpecEntry(ParameterCode.parseParameterCode(parameterCode), ColumnMetadata.SpecEntry.SpecType.PARAM)));
+		log.debug("Instantaneous ancillary columns constructed : " + result.keySet().toString());
```
Comment (davidmsibley) [inline gcmrc-services/src/main/java/gov/usgs/cida/gcmrcservices/column/ColumnResolver.java]: This looks about right.  We may be able to move from hardcoded "Discharge" and "Stage" in the future.

**Comment by davidmsibley** [comment]
Mark, this looks great, thanks for working on this.  I've got some things we can walk through next week for optimizing the java code.

File: gcmrc-services/src/main/java/gov/usgs/cida/gcmrcservices/column/ColumnMetadata.java [hunk 3]
```diff
@@ -89,10 +89,38 @@ private static String hashString(String message, Integer places) {
 	
 	public static String createColumnName(String station, ParameterCode parameterCode) {
 		String result = null;
-		
-		if (null != station && null != parameterCode) {
-			result = "S" + hashString(station, 5) + "P" + hashString(parameterCode.toString(), 5);
+		String ancillaryColumn = null;
+		ParameterCode tempParameterCode = null;
+
+		switch (parameterCode.sampleMethod+parameterCode.groupName)
```
Comment (davidmsibley) [inline gcmrc-services/src/main/java/gov/usgs/cida/gcmrcservices/column/ColumnMetadata.java]: do we have to switch here on groupName? either way we're just setting tempParameterCode to be the inst sampleMethod and whatever the existing groupName was.

Comment (mamcderm-usgs) [inline gcmrc-services/src/main/java/gov/usgs/cida/gcmrcservices/column/ColumnMetadata.java]: @davidmsibley - no, thanks for catching.  does this look right now?  should I get rid of that sampleMethod check somehow...

Comment (davidmsibley) [inline gcmrc-services/src/main/java/gov/usgs/cida/gcmrcservices/column/ColumnMetadata.java]: looks better. Yeah that sampleMethod check would be prime for moving into some translation datastructure, but it's alright for now.

**Comment by davidmsibley** [comment]
:+1:

**Pull request merged**